### ISSUE TITLE
LoginDialog not cleaning it's widgets properly causing lag.

### ISSUE THREAD

**Issue by onde2rock** [koreader/koreader#3797]
* KOReader version:git
* Device:SDL

#### Issue
When clicking alternatively on both text input of a login dialog, koreader start to lag. 
#### Steps to reproduce
Click alternatively on both text input of a login dialog,
##### `crash.log` (if applicable)
log after clicking maybe ten times : 

```
03/24/18-23:12:12 DEBUG AutoSuspend: onInputEvent
03/24/18-23:12:12 DEBUG close widget
03/24/18-23:12:12 DEBUG show widget unknown
03/24/18-23:12:12 DEBUG show widget unknown
03/24/18-23:12:12 DEBUG TextBoxWidget:free called
03/24/18-23:12:12 DEBUG painting widget: ReaderUI
03/24/18-23:12:12 DEBUG painting widget: table: 0x4190e100
03/24/18-23:12:12 DEBUG painting widget: table: 0x40abb700
03/24/18-23:12:12 DEBUG painting widget: table: 0x41907818
03/24/18-23:12:12 DEBUG painting widget: table: 0x4178df70
03/24/18-23:12:12 DEBUG painting widget: table: 0x4065dfb0
03/24/18-23:12:12 DEBUG painting widget: table: 0x40bca068
03/24/18-23:12:12 DEBUG painting widget: table: 0x412982f8
03/24/18-23:12:12 DEBUG painting widget: table: 0x401818d8
03/24/18-23:12:12 DEBUG painting widget: table: 0x40f8ecd8
03/24/18-23:12:12 DEBUG painting widget: table: 0x40880ce8
03/24/18-23:12:12 DEBUG painting widget: table: 0x415d61b8
03/24/18-23:12:12 DEBUG painting widget: table: 0x4117fe80
03/24/18-23:12:12 DEBUG painting widget: table: 0x41195720
03/24/18-23:12:12 DEBUG painting widget: table: 0x41870898
03/24/18-23:12:12 DEBUG painting widget: table: 0x41795678
03/24/18-23:12:13 DEBUG painting widget: table: 0x4131f708
03/24/18-23:12:13 DEBUG painting widget: table: 0x41ea7c28
03/24/18-23:12:13 DEBUG painting widget: table: 0x4053f478
03/24/18-23:12:13 DEBUG blitFrom 59 1336 0 0 48 48
03/24/18-23:12:13 DEBUG blitFrom 892 1336 0 0 48 48
03/24/18-23:12:13 DEBUG blitFrom 182 1424 0 0 48 48
03/24/18-23:12:13 DEBUG blitFrom 892 1424 0 0 48 48
03/24/18-23:12:13 DEBUG painting widget: table: 0x4053f6a8
03/24/18-23:12:13 DEBUG refresh on physical rectangle -1 1132 1002 369
03/24/18-23:12:13 DEBUG refresh on physical rectangle 21 417 958 298
03/24/18-23:12:14 DEBUG BackgroundRunner: _execute() @  1521929534
03/24/18-23:12:14 DEBUG BackgroundRunnerWidget: start running @  1521929534
03/24/18-23:12:15 DEBUG key event => type: 1, code: 226(Alt), value: 1, time: 1521929535.418756
03/24/18-23:12:15 DEBUG key event => type: 1, code: 61(F4), value: 1, time: 1521929535.418769
03/24/18-23:12:15 DEBUG refresh on physical rectangle 0 0 1000 1500
03/24/18-23:12:15 INFO  initializing for device SDL
03/24/18-23:12:15 INFO  framebuffer resolution: {
    ["h"] = 1500,
    ["w"] = 1000
}
03/24/18-23:12:15 DEBUG close widget
03/24/18-23:12:15 DEBUG AutoSuspend: onInputEvent
03/24/18-23:12:15 DEBUG closing reader
03/24/18-23:12:15 DEBUG closing document
# 03/24/18-23:12:15  on close document
03/24/18-23:12:15 DEBUG close widget ReaderUI
03/24/18-23:12:15 DEBUG save default dictionary
03/24/18-23:12:15 DEBUG Rename  /home/aurel/git/koreader/koreader/test/juliet.sdr/metadata.epub.lua  to  /home/aurel/git/koreader/koreader/test/juliet.sdr/metadata.epub.lua.old
03/24/18-23:12:15 DEBUG Write to  /home/aurel/git/koreader/koreader/test/juliet.sdr/metadata.epub.lua
03/24/18-23:12:15 DEBUG show widget unknown
03/24/18-23:12:15 DEBUG painting widget: table: 0x4190e100
03/24/18-23:12:15 DEBUG painting widget: table: 0x40abb700
03/24/18-23:12:15 DEBUG painting widget: table: 0x41907818
03/24/18-23:12:15 DEBUG painting widget: table: 0x4178df70
03/24/18-23:12:15 DEBUG painting widget: table: 0x4065dfb0
03/24/18-23:12:15 DEBUG painting widget: table: 0x40bca068
03/24/18-23:12:15 DEBUG painting widget: table: 0x412982f8
03/24/18-23:12:15 DEBUG painting widget: table: 0x401818d8
03/24/18-23:12:15 DEBUG painting widget: table: 0x40f8ecd8
03/24/18-23:12:15 DEBUG painting widget: table: 0x40880ce8
03/24/18-23:12:15 DEBUG painting widget: table: 0x415d61b8
03/24/18-23:12:15 DEBUG painting widget: table: 0x4117fe80
03/24/18-23:12:15 DEBUG painting widget: table: 0x41195720
03/24/18-23:12:15 DEBUG painting widget: table: 0x41870898
03/24/18-23:12:15 DEBUG painting widget: table: 0x41795678
03/24/18-23:12:16 DEBUG painting widget: table: 0x4131f708
03/24/18-23:12:16 DEBUG painting widget: table: 0x41ea7c28
03/24/18-23:12:16 DEBUG painting widget: table: 0x4053f478
03/24/18-23:12:16 DEBUG blitFrom 59 1336 0 0 48 48
03/24/18-23:12:16 DEBUG blitFrom 892 1336 0 0 48 48
03/24/18-23:12:16 DEBUG blitFrom 182 1424 0 0 48 48
03/24/18-23:12:16 DEBUG blitFrom 892 1424 0 0 48 48
03/24/18-23:12:16 DEBUG painting widget: table: 0x4053f6a8
03/24/18-23:12:16 DEBUG painting widget: table: 0x40bd06a8
03/24/18-23:12:16 DEBUG refresh on physical rectangle -1 -1 1002 1502
03/24/18-23:12:16 DEBUG BackgroundRunner: _execute() @  1521929536
03/24/18-23:12:16 DEBUG BackgroundRunnerWidget: start running @  1521929536
03/24/18-23:12:16 INFO  quitting uimanager
03/24/18-23:12:16 DEBUG close widget
03/24/18-23:12:16 INFO  no dialog left to show
```

**Comment by onde2rock** [comment]
Find it, it's because of this line which need to go : 

https://github.com/koreader/koreader/blob/9443098e98917c94a3a8c7e4e9d1b84bce1928cf/frontend/ui/widget/multiinputdialog.lua#L135

**Comment by Frenzie** [comment]
Included in #3796.
